bentoml/yatai/deployment/aws_ec2/operator.py

```diff
@@ -0,0 +1,559 @@
+        or autoscaling_desired_size > autoscaling_max_size
+    ):
+        raise BentoMLException(
+            "Wrong autoscaling capacity specified. "
+            "It should be min_size <= desired_size <= max_size"
+        )
+
+    template_file_path = os.path.join(project_dir, sam_template_name)
+    yaml = YAML()
+    sam_config = {
+        "AWSTemplateFormatVersion": "2010-09-09",
+        "Transform": "AWS::Serverless-2016-10-31",
+        "Description": "BentoML load balanced template template",
+        "Parameters": {
+            "AmazonLinux2LatestAmiId": {
+                "Type": "AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>",
+                "Default": ami_id,
+            },
+        },
+    }
+    yaml.dump(sam_config, Path(template_file_path))
+
+    with open(template_file_path, "a") as f:
+        f.write(
+            """\
+Resources:
+    SecurityGroupResource:
+        Type: AWS::EC2::SecurityGroup
+        Properties:
+            GroupDescription: "security group for bentoservice"
+            SecurityGroupIngress:
+                -
+                    IpProtocol: tcp
+                    CidrIp: 0.0.0.0/0
+                    FromPort: 5000
+                    ToPort: 5000
+                -
+                    IpProtocol: tcp
+                    CidrIp: 0.0.0.0/0
+                    FromPort: 22
+                    ToPort: 22
+
+    LaunchTemplateResource:
+        Type: AWS::EC2::LaunchTemplate
+        Properties:
+            LaunchTemplateName: {template_name}
+            #Key and security gorups remainign for logging in
+            LaunchTemplateData:
+                ImageId: !Ref AmazonLinux2LatestAmiId
+                InstanceType: {instance_type}
+                UserData: "{user_data}"
+                SecurityGroupIds:
+                - !GetAtt SecurityGroupResource.GroupId
+
+    AutoScalingGroup:
+        Type: AWS::AutoScaling::AutoScalingGroup
+        Properties:
+            MinSize: {autoscaling_min_size}
+            MaxSize: {autoscaling_max_size}
+            DesiredCapacity: {autoscaling_desired_size}
+            AvailabilityZones: !GetAZs
+            LaunchTemplate:
+                LaunchTemplateId: !Ref LaunchTemplateResource
+                Version: !GetAtt LaunchTemplateResource.LatestVersionNumber
+Outputs:
+    S3Bucket:
+        Value: {s3_bucket_name},
+        Description: 'S3 Bucket for saving artifacts and lambda bundle'
+""".format(
+                template_name=sam_template_name,
+                instance_type=instance_type,
+                user_data=user_data,
+                autoscaling_min_size=autoscaling_min_size,
+                autoscaling_desired_size=autoscaling_desired_size,
+                autoscaling_max_size=autoscaling_max_size,
+                s3_bucket_name=s3_bucket_name,
+            )
+        )
+    return template_file_path
+
+
+class AwsEc2DeploymentOperator(DeploymentOperatorBase):
+    def deploy_service(
```

Comment:
Good job refactoring out the deploy code!

Reply:
Thanks